otter/models/interface.py

```diff
@@ -3,89 +3,83 @@
 """
 from datetime import datetime
 
+import attr
+from attr.validators import instance_of as aiof
+
 from croniter import croniter
 
+import six
+
 from twisted.python.constants import NamedConstant, Names
 
 from zope.interface import Attribute, Interface
 
 from otter.util import timestamp
+from otter.util.http import text
 
 
+class ScalingGroupStatus(Names):
+    """
+    Status of scaling group
+    """
+
+    ACTIVE = NamedConstant()
+    "Group is active and executing policies/converging"
+
+    ERROR = NamedConstant()
+    """
+    Group has errored due to (mostly) invalid launch configuration and has
+    stopped executing policies/converging
+    """
+
+    DELETING = NamedConstant()
+    """
+    Group is getting deleted and all it's resources servers/CLBs
+    are getting deleted
+    """
+
+
+@attr.s
 class GroupState(object):
     """
     Object that represents the state
 
     :ivar bytes tenant_id: the tenant ID of the scaling group whose state this
         object represents
-
     :ivar bytes group_id: the ID of the scaling group whose state this
         object represents
-    :ivar bytes group_name: the name of the scaling group whose state this
```

Comment:
This should still have type information associated with it.  Given the intent of the rest of the PR, it should probably be`text_type` rather than `bytes`.